namer/consul/src/test/scala/io/buoyant/namer/consul/ConsulNamerTest.scala

```diff
@@ -503,13 +503,76 @@ class ConsulNamerTest extends FunSuite with Awaits {
     @volatile var state: Activity.State[NameTree[Name]] = Activity.Pending
     namer.lookup(Path.read("/dc1/servicename/residual")).states respond { state = _ }
 
-    assertOnAddrs(state) { (_, metadata) =>
-      assert(metadata == Addr.Metadata(Metadata.authority -> "servicename.service.dc1.consul.acme.co"))
-      ()
+    assert(state == Activity.Ok(NameTree.Neg))
+  }
+
+  test("Namer state returns Neg then Bound then Neg when a datacenter becomes available and then becomes unavailable") {
+    val datacenterWillBeUnavailable = new Promise[Unit]
+    val datacenterIsAvailable = new Promise[Unit]
+    val scenarioComplete = new Promise[Unit]
+    @volatile var datacenterIsUp = false
+    class TestApi extends CatalogApi(null, "/v1") {
+      override def serviceNodes(
+         serviceName: String,
+         datacenter: Option[String],
+         tag: Option[String],
+         blockingIndex: Option[String],
+         consistency: Option[ConsistencyMode],
+         retry: Boolean
+       ): Future[Indexed[Seq[ServiceNode]]] = blockingIndex match {
+        case Some("0") | None =>
+          if (datacenterIsUp)
+            datacenterIsAvailable before Future.value(
+              Indexed[Seq[ServiceNode]](Seq(testServiceNode, testServiceNode2), Some("1"))
+            )
+          else {
+            datacenterIsUp = true
+            Future.exception(new Exception("datacenter does not exist yet"))
+          }
+        case Some("1") =>
+          datacenterWillBeUnavailable before Future.exception(new Exception("something bad happened again"))
+        case _ =>
+          scenarioComplete.setDone()
+          Future.never
+      }
     }
 
+    val stats = new InMemoryStatsReceiver
+    val namer = ConsulNamer.untagged(
+      Path.read("/test"),
+      new TestApi(),
+      new TestAgentApi("consul.acme.co"),
+      setHost = false,
+      stats = stats
+    )
+    @volatile var state: Activity.State[NameTree[Name]] = Activity.Pending
+    @volatile var nameTreeUpdates: Int = 0
+    namer.lookup(Path.read("/#/io.l5d.consul/dc1/servicename/residual")).states respond { s =>
+      nameTreeUpdates += 1
+      log.info(s"count: $nameTreeUpdates")
+      state = s
+    }
+
+
+    withClue("before datacenter is available") {
+      assert(state == Activity.Ok(NameTree.Neg))
```

Comment:
I'm not sure, but you might want to wrap this in eventually as well.  Depending on the timing I think this could potentially still be pending